pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -951,14 +958,23 @@ func (m *kubeGenericRuntimeManager) killPodWithSyncResult(pod *v1.Pod, runningPo
 	// stop sandbox, the sandbox will be removed in GarbageCollect
 	killSandboxResult := kubecontainer.NewSyncResult(kubecontainer.KillPodSandbox, runningPod.ID)
 	result.AddSyncResult(killSandboxResult)
+
+	wg := sync.WaitGroup{}
+	wg.Add(len(runningPod.Sandboxes))
 	// Stop all sandboxes belongs to same pod
 	for _, podSandbox := range runningPod.Sandboxes {
-		if err := m.runtimeService.StopPodSandbox(podSandbox.ID.ID); err != nil {
-			killSandboxResult.Fail(kubecontainer.ErrKillPodSandbox, err.Error())
-			klog.ErrorS(nil, "Failed to stop sandbox", "podSandboxID", podSandbox.ID)
-		}
+		go func(podSandbox *kubecontainer.Container) {
+			defer wg.Done()
+			if err := m.runtimeService.StopPodSandbox(podSandbox.ID.ID); err != nil {
+				killSandboxResult.Fail(kubecontainer.ErrKillPodSandbox, err.Error())
+				klog.ErrorS(nil, "Failed to stop sandbox", "podSandboxID", podSandbox.ID)
+			} else if m.statusManager != nil {
+				m.statusManager.SetContainerTerminated(pod.UID, podSandbox.ID)
+				klog.Info("Stop sandbox", "podSandboxID", podSandbox.ID)
```

Comment:
can you use structured logging here? https://kubernetes.io/blog/2020/09/04/kubernetes-1-19-introducing-structured-logs/

Reply:
Thanks, updated.